### PR TITLE
Feature/29 admin merchant create

### LINKED ISSUE #35: feature/29-admin-merchant-create
29. Admin Merchant Create

As an admin,
When I visit the admin merchants index (/admin/merchants)
I see a link to create a new merchant.
When I click on the link,
I am taken to a form that allows me to add merchant information.
When I fill out the form I click ‘Submit’
Then I am taken back to the admin merchants index page
And I see the merchant I just created displayed
And I see my merchant was created with a default status of disabled.

### PR DESCRIPTION
**Pull request by aplasket**
## Checklist before requesting a review
- [x] Verified Render deploy is functional
- [x] Pulled/merged main 
- [x] Tests for the changes have been added into spec files
- [x] Branch name follows our guidelines 
- [x] All tests running clean
- [x] Ran this in Rails S  - features are working properly
- [x] Test coverage is at 100% for models and features

## What changes or new behavior does this PR introduce (Bug fix, feature, update, etc)?
  - add link and form to add new merchant
  - add flash messages and sad path testing 
  - add validation for name to merchant model and model test
  - ensure when new merchant is added, default status is disabled
 
## Bugs, Issues, etc:
 - [ ] check if there are known bugs or issue

### CODE REVIEW

**Review by kbergstrom78 — APPROVED** [review]
I feel like we're living in parallel coding universes right now.  This looks great!

File: app/models/merchant.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Merchant < ApplicationRecord
   has_many :items
+  validates_presence_of :name
```
Comment (AlexKiselich) [inline app/models/merchant.rb:3]: validations, gotta love it

File: spec/features/admin/merchants/new_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      within "#disabled-merchants" do
+        expect(page).to have_content("Bob's Burgers")
+      end
+    end
+
+    it "displays a flash message if name is not inputted" do
+      visit new_admin_merchant_path
+      expect(page).to have_content("New Merchant")
+
+      fill_in "Name", with: ""
```
Comment (AlexKiselich) [inline spec/features/admin/merchants/new_spec.rb:23]: Sad path, very nice!!

**Review by AlexKiselich — APPROVED** [review]
Great work, as usual! This was fun testing out on my computer too haha

**Pull request merged**